views/js/qtiCreator/helper/xincludeLoader.js

```diff
@@ -0,0 +1,105 @@
+                            reject(new Error('File not found'));
+                        });
+                    } else {
+                        reject(new Error('href or baseUrl is missing'));
+                    }
+                });
+            },
+            parseXmlToDom: function parseXmlToDom(xmlString) {
+                // Parse the XML string into a document object
+                const parser = new DOMParser();
+                const xmlDoc = parser.parseFromString(xmlString, "application/xml");
+
+                // Function to recursively convert XML to HTML
+                function convertXMLToHTML(xmlNode) {
+                    // Create a corresponding HTML element for the XML node
+                    const htmlNode = document.createElement(xmlNode.nodeName);
+
+                    // Copy attributes
+                    for (let i = 0; i < xmlNode.attributes.length; i++) {
+                        const attr = xmlNode.attributes[i];
+                        htmlNode.setAttribute(attr.name, attr.value);
+                    }
+
+                    // Recursively convert child nodes
+                    xmlNode.childNodes.forEach(childNode => {
+                        if (childNode.nodeType === Node.ELEMENT_NODE) {
+                            htmlNode.appendChild(convertXMLToHTML(childNode));
+                        } else if (childNode.nodeType === Node.TEXT_NODE) {
+                            htmlNode.appendChild(document.createTextNode(childNode.nodeValue));
+                        }
+                    });
+
+                    return htmlNode;
+                }
+
+                // Convert the XML document to HTML
+                return convertXMLToHTML(xmlDoc.documentElement);
+            },
+            loadByElementPages: function loadByElementPages(pages, baseUrl) {
+                return pages.map(page => {
+                    const contentPromises = page.content.map(contentItem => {
+                        const tempDiv = document.createElement('div');
+                        tempDiv.innerHTML = contentItem;
+                        const xiIncludeElements = tempDiv.querySelectorAll('xi\\:include');
+
+                        const xiIncludePromises = Array.from(xiIncludeElements).map(xiIncludeElement => {
+                            const $xiIncludeElement = $(xiIncludeElement);
+                            return xincludeLoader.load($xiIncludeElement, baseUrl).then(newContent => {
+                                $xiIncludeElement.replaceWith(newContent.data);
+                            });
+                        });
+
+                        return Promise.all(xiIncludePromises).then(() => tempDiv.innerHTML);
+                    });
+
+                    return Promise.all(contentPromises).then(updatedContentItems => {
+                        page.content = updatedContentItems;
+                        return page;
+                    });
+                });
```

Comment:
**question:** Is this code specific to a particular PCI??? If yes, please make it generic, or move it to the PCI code